src/are-inputs-equal.ts

```diff
@@ -10,10 +10,28 @@ export default function areInputsEqual(
   // https://github.com/alexreardon/memoize-one/pull/59
 
   for (let i = 0; i < newInputs.length; i++) {
-    // using shallow equality check
-    if (newInputs[i] !== lastInputs[i]) {
+    if (hasChanged(newInputs[i], lastInputs[i])) {
       return false;
     }
   }
   return true;
 }
+
+// compare whether a value has changed, accounting for NaN
+function hasChanged(newValue: unknown, oldValue: unknown): boolean {
+  if (newValue === oldValue) {
+    return false;
+  }
+
+  // Special case for NaN. NaN is not === to NaN
+  if (isNaN(newValue) && isNaN(oldValue)) {
+    return false;
+  }
+
+  return true;
+}
+
+function isNaN(value: unknown) {
+  // check for NaN, by comparing value to itself.
+  return value !== value;
+}
```

Comment:
I created a custom `isNaN` function because `Number.isNaN` [isn't supported in ie11](https://caniuse.com/es6-number).
Or this library doesn't support older browsers?